cwms-data-api/src/main/java/cwms/cda/api/MeasurementController.java

```diff
@@ -0,0 +1,247 @@
+ * SOFTWARE.
+ */
+package cwms.cda.api;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.MetricRegistry;
+import static com.codahale.metrics.MetricRegistry.name;
+import com.codahale.metrics.Timer;
+import static cwms.cda.api.Controllers.CREATE;
+import static cwms.cda.api.Controllers.DELETE;
+import static cwms.cda.api.Controllers.FAIL_IF_EXISTS;
+import static cwms.cda.api.Controllers.GET_ALL;
+import static cwms.cda.api.Controllers.GET_ONE;
+import static cwms.cda.api.Controllers.ID_MASK;
+import static cwms.cda.api.Controllers.MAX_DATE;
+import static cwms.cda.api.Controllers.MIN_DATE;
+import static cwms.cda.api.Controllers.NOT_SUPPORTED_YET;
+import static cwms.cda.api.Controllers.MIN_NUMBER;
+import static cwms.cda.api.Controllers.MAX_NUMBER;
+import static cwms.cda.api.Controllers.OFFICE;
+import static cwms.cda.api.Controllers.OFFICE_MASK;
+import static cwms.cda.api.Controllers.UNIT_SYSTEM;
+import static cwms.cda.api.Controllers.requiredParam;
+import cwms.cda.api.enums.UnitSystem;
+import cwms.cda.data.dao.MeasurementDao;
+import cwms.cda.data.dto.measurement.Measurement;
+import cwms.cda.formatters.ContentType;
+import cwms.cda.formatters.Formats;
+import cwms.cda.formatters.FormattingException;
+import io.javalin.apibuilder.CrudHandler;
+import io.javalin.core.util.Header;
+import io.javalin.http.Context;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiRequestBody;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.time.Instant;
+import java.util.ArrayList;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.DSLContext;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+
+import static cwms.cda.data.dao.JooqDao.getDslContext;
+
+public final class MeasurementController implements CrudHandler {
+
+    static final String TAG = "Measurements";
+
+    private final MetricRegistry metrics;
+    private final Histogram requestResultSize;
+
+    public MeasurementController(MetricRegistry metrics) {
+        this.metrics = metrics;
+        String className = this.getClass().getName();
+        requestResultSize = this.metrics.histogram(name(className, "results", "size"));
+    }
+
+    private Timer.Context markAndTime(String subject) {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
+    }
+
+    @OpenApi(
+            queryParams = {
+                    @OpenApiParam(name = OFFICE_MASK, description = "Office id for filtering measurements."),
+                    @OpenApiParam(name = ID_MASK, description = "Location id for filtering measurements."),
+                    @OpenApiParam(name = MIN_NUMBER, description = "Minimum number for filtering measurements."),
```

Comment:
Would these be the min and max of the measurement value? I feel that the description could be a bit clearer as to what these are filtering on.

Reply:
For measurements, there is a number field which isn't really a number. Its a String value that acts as an identifier, but there is order to it. 

Normally it is a number like "1", "2", etc.. but it could be "1A", "1.B", etc. 

I updated the description to use "number-id" to be more clear.